internal/repositories/devicedefinition/devicedefinition.go

```diff
@@ -0,0 +1,224 @@
+package devicedefinition
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/internal/repositories/base"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+	"golang.org/x/exp/slices"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+// TokenPrefix is the prefix for a global token id for Device Definition.
+const TokenPrefix = "DD"
+
+type DeviceDefinitionRepository struct {
+	*base.Repository
+}
+
+func ToAPI(v *DeviceDefinitionTablelandModel) *gmodel.DeviceDefinition {
+	var result = gmodel.DeviceDefinition{}
+
+	result.ID = v.ID
+	result.Ksuid = &v.KSUID
+	result.Year = &v.Year
+	result.Model = &v.Model
+
+	if v.Metadata.DeviceAttributes != nil && len(v.Metadata.DeviceAttributes) > 0 {
+		for _, attr := range v.Metadata.DeviceAttributes {
+			result.Attributes = append(result.Attributes, &gmodel.DeviceDefinitionAttribute{
+				Name:  &attr.Name,
+				Value: &attr.Value,
+			})
+		}
+	}
+
+	return &result
+}
+
+func (r *DeviceDefinitionRepository) GetDeviceDefinition(ctx context.Context, by gmodel.DevicedefinitionBy) (*gmodel.DeviceDefinition, error) {
+	if by.Manufacturer == nil || *by.Manufacturer == "" {
+		return nil, gqlerror.Errorf("Provide exactly one `manufacturer`.")
+	}
+
+	if by.ID == nil || *by.ID == "" {
+		return nil, gqlerror.Errorf("Provide exactly one `ID`.")
+	}
+
+	tableName := "_80001_8439"
+	statement := fmt.Sprintf("SELECT * FROM %s WHERE id = '%s'", tableName, *by.ID)
```

Comment:
Do you know how tableland handles query injection because right now this is unsafe because the value for by.ID could be `"1; DROP TABLES;`